### ISSUE TITLE
using test set as validation set

### ISSUE THREAD

**Issue by aresreact** [dmlc/xgboost#9235]
While doing a walk-forward validation to assess the performances of my model, I specified the test set as validation set in the following way:

```
    booster = xgb.train(params,
                dmat_train,
                evals=[(dmat_train, "train"), (dmat_test, "test")], verbose_eval = False,
                num_boost_round=num_boost_round)
    preds = booster.predict(dmat_test

)
```

thinking that it would have no effect and would be only for metrics monitoring during training.

However, when I do not do it:

```
    booster = xgb.train(params,
                dmat_train,
                evals=[(dmat_train, "train")], verbose_eval = False,
                num_boost_round=num_boost_round)
    preds = booster.predict(dmat_test)
```

performances are significantly worse.

I guess the second way of doing is the right one. But why? How does specifying the test set as cross-validation set actually leads to overfitting in XGBoost? Note that I do not do any hyper-parameter optimizations.

**Comment by trivialfis** [comment]
Could you please share the parameters that you are using?
